bisection/application.py

```diff
@@ -74,6 +75,9 @@ def collect_programs(threshold_lines):
             os.path.join(directory, '../backend')
     ]:
         for root, _, files in os.walk(top):
+            # Ignore all ignored directories
+            if [x for x in IGNORE_DIRECTORIES if x in root]:
```

Comment:
Can you make IGNORE_DIRECTORIES a set and then check if root in IGNORE_DIRECTORIES?

Reply:
`root` can be something like "frontend/node_modules/xx", so I have to check if an entry in `IGNORE_DIRECTORIES` is a substring of `root` or not.

Reply:
ahh got it, thanks